pkgs/tools/misc/nbqa/default.nix

```
@@ -0,0 +1,93 @@
+{ black
+, blacken-docs
+, fetchFromGitHub
+, lib
+, python3
+, ruff
+}:
+python3.pkgs.buildPythonApplication rec {
+  pname = "nbqa";
+  version = "1.7.0";
+
+  src = fetchFromGitHub {
+    owner = "nbQA-dev";
+    repo = "nbQA";
+    rev = version;
+    hash = "sha256-CTF5HisBS44Ta18cVT04UrMIF30WmEBwZBGW7fkKXwk=";
+  };
+
+  passthru.optional-dependencies = {
+    # Based on https://github.com/nbQA-dev/nbQA/blob/1.7.0/setup.cfg#L47-L56
+    black = [ black ];
+    blacken-docs = [ blacken-docs ];
+    flake8 = [ python3.pkgs.flake8 ];
+    isort = [ python3.pkgs.isort ];
+    jupytext = [ python3.pkgs.jupytext ];
+    mypy = [ python3.pkgs.mypy ];
+    pylint = [ python3.pkgs.pylint ];
+    pyupgrade = [ python3.pkgs.pyupgrade ];
+    ruff = [ ruff ];
+  };
+
+  propagatedBuildInputs = with python3.pkgs;
+    [
+      # Based on https://github.com/nbQA-dev/nbQA/blob/1.7.0/setup.cfg#L31-L35
+      autopep8
+      ipython
+      tokenize-rt
+      tomli
+    ]
+    ++ builtins.attrValues passthru.optional-dependencies;
+
+  postPatch = ''
+    # Force using the Ruff executable rather than the Python package
+    sed --in-place --expression='s/if shell:/if shell or main_command == "ruff":/' nbqa/__main__.py
+  '';
+
+  preCheck = ''
+    # Allow the tests to run `nbqa` itself from the path
+    export PATH="$out/bin":"$PATH"
+  '';
+
+  checkInputs = with python3.pkgs; [
+    autoflake
+    flake8
+    isort
+    jupytext
+    mdformat
+    pre-commit-hooks
+    pydocstyle
+    pylint
+    pytestCheckHook
+    pyupgrade
+    yapf
+  ];
+
+  nativeCheckInputs = [
+    black
+    ruff
+  ];
+
+  disabledTests = [
+    "test_black_multiple_files" # Unknown why this is failing
+    "test_black_return_code" # Unknown why this is failing
+    "test_grep" # Unknown why this is failing
+    "test_jupytext_on_folder" # Unknown why this is failing
+    "test_mypy_works" # Unknown why this is failing
+    "test_running_in_different_dir_works" # Unknown why this is failing
+    "test_unable_to_reconstruct_message_pythonpath" # Unknown why this is failing
+    "test_with_subcommand" # Unknown why this is failing
+  ];
+
+  disabledTestPaths = [
+    "tests/test_include_exclude.py" # Unknown why these are failing
```

```suggestion
    # test data not found
    "test_black_multiple_files"
    "test_black_return_code"
    "test_grep"
    "test_jupytext_on_folder"
    "test_mypy_works"
    "test_running_in_different_dir_works"
    "test_unable_to_reconstruct_message_pythonpath"
    "test_with_subcommand"
  ];

  disabledTestPaths = [
    # test data not found
    "tests/test_include_exclude.py"
```

Please describe the error.

---

Thank you!